### PR TITLE
Migrate some JUnit 4 tests to JUnit 5

### PR DESCRIPTION
**Pull request by sh-cho**
Motivation:

JUnit5 is recommended over JUnit 4 for several reasons

Modifications:

- Migrate JUnit 4 tests to JUnit 5
- Clean up
  - Removed `@Nonnull` annotations, since armeria is non-null by default.

Result:

Migrated to JUnit 5

---

Hi, I'm trying to work on a JUnit 4 -> 5 migration.

~It's going to take some time to migrate all tests, so I'd like to make it as draft and going to open it when it's done, is that ok?~
-> I'm working on a separate branch :)

### CODE REVIEW

**Review by trustin — APPROVED** [review]
Thanks, @sh-cho! Looks good to me. :bow:

**Comment by jrhee17** [comment]
> It's going to take some time to migrate all tests, so I'd like to make it as draft and going to open it when it's done, is that ok?

I think it's fine to mark this PR as ready now and get the current changeset merged.
Alternatively, I think it's also fine to keep going and migrate other tests you would like.

File: core/src/test/java/com/linecorp/armeria/internal/common/metric/CaffeineMetricSupportTest.java [hunk 1]
```diff
@@ -47,21 +43,19 @@
 
 import io.micrometer.core.instrument.MeterRegistry;
 
+@ExtendWith(MockitoExtension.class)
 public class CaffeineMetricSupportTest {
```
Comment (jrhee17) [inline core/src/test/java/com/linecorp/armeria/internal/common/metric/CaffeineMetricSupportTest.java:47]: `junit5` test classes don't have to be public
```suggestion
class CaffeineMetricSupportTest {
```

**Review by jrhee17 — APPROVED** [review]
The current changeset looks good bar one comment 👍

**Comment by sh-cho** [comment]
> > It's going to take some time to migrate all tests, so I'd like to make it as draft and going to open it when it's done, is that ok?
> 
> I think it's fine to mark this PR as ready now and get the current changeset merged. Alternatively, I think it's also fine to keep going and migrate other tests you would like.

Ok. I'm going to make this PR open and will work on another branch 😃

**Review by ikhoon — APPROVED** [review]
Thanks, @sh-cho!

**Review by minwoox — APPROVED** [review]
Great, thanks!

**Pull request merged**